runtime/interpreter/interpreter.go

```diff
@@ -3481,34 +3481,40 @@ func referenceTypeFunction(invocation Invocation) Value {
 	}
 
 	var authorization Authorization = UnauthorizedAccess
-	var entitlements []common.TypeID = make([]common.TypeID, 0, entitlementValues.Count())
 	errInIteration := false
 
-	entitlementValues.Iterate(invocation.Interpreter, func(element Value) (resume bool) {
-		entitlementString, isString := element.(*StringValue)
-		if !isString {
-			errInIteration = true
-			return false
-		}
+	if entitlementValues.Count() > 0 {
```

Comment:
Refactor repeated expression into variable and reuse in the two other places below:

```suggestion
	entitlementsCound := entitlementValues.Count()
	if entitlementsCound > 0 {
```